### ISSUE TITLE
Vertical switch animation even when moving horizontally

### ISSUE THREAD

**Issue by amcsi** [zakkak/workspace-grid#104]
I'm using Ubuntu 18.10, and with workspace-grid set up to do 2x2 workspaces, I'm getting a vertical scroll animation even if I scroll horizontally.

How can I fix this? It's also sufficient if I could just disable animations altogether, as long as this visually confusing phenomenon does not happen.

**Comment by corund** [comment]
I have the same phenomenon. I'm using gnome-shell 3.30.1-2, debian buster (10), grid set up 3x3.

**Comment by hkdb** [comment]
There's an issue with mutter that has already been fixed and will take a bit of time for it to make it downstream. You can find the workaround for now here: https://github.com/zakkak/workspace-grid/issues/88#issuecomment-435037672

**Comment by hkdb** [comment]
In case this helps someone else, I put together a step by step for both Arch and Ubuntu 18.10: https://medium.com/@hkdb/patch-gnome-3-30-for-workspace-grid-1ee8b1dd92f9

**Comment by txelu** [comment]
I could make it work in Arch Linux with the instructions by hkdb. Thanks!

**Comment by alexykot** [comment]
Can confirm, am affected on Arch as well, and hkdb patch helped me as well.

**Comment by harish2704** [comment]
I think this mutter bug  https://gitlab.gnome.org/GNOME/mutter/issues/409 
is the root cause of this issue.

**Comment by alexykot** [comment]
I see you've closed your bug submission in Mutter. I assume some of the upcoming releases will ship that fix to repos, right?

**Comment by harish2704** [comment]
@alexykot : yes. I found that this bug is already fixed in master branch.

**Comment by LDVSOFT** [comment]
Can someone help to update patch? Seems like Ubuntu 18.10 now uses 3.30.2, and it requires another patch...

**Comment by alexykot** [comment]
Few days ago I've applied this patch to Gnome 3.30.2 on Arch and had no issues. What is the exact problem you're facing?

**Comment by LDVSOFT** [comment]
@alexykot It's not helping. Built the debs, installed, rebooted: still only up-down animations.

**Comment by jonnixs** [comment]
There seems like there may have been a regression with Mutter, but I don't know enough about the project to raise an issue on it.

**Comment by alexykot** [comment]
Earlier in this thread it was discussed already. Bug submission is https://gitlab.gnome.org/GNOME/mutter/issues/409, and it was already fixed in `master` but not made it into a release yet. 

Or do you see a new bug there?

**Comment by jonnixs** [comment]
I had originally tested the patch when it was added into master and it worked as expected, but having kept mutter updated, the error is there. I don't think workspace-grid has changed in that time, so my current assumption is there's a regression, though the issue might be elsewhere (or simply with my machine, though it seems like others are having the same problem).

The [PR](https://gitlab.gnome.org/GNOME/mutter/merge_requests/245) related to that issue was merged into master pretty quickly, and then released in version `3.29.90` according to the [release notes](https://gitlab.gnome.org/GNOME/mutter/blob/master/NEWS). Ubuntu appears to now be on `3.30.2` so should already have the patch included.

**Comment by alexykot** [comment]
Right, I see. Yes, Arch is on `3.30.2` as well, and the problem persists here as well. Also the patch is still working for it, so I still can fix it with the help of this thread. 

Still, seems like a new mutter bugreport is needed here. To be honest I'm not in the know as well to produce a meaningful submission.

**Comment by LDVSOFT** [comment]
Again, as I said, patch from Arch does not help for me :/ If anyone would upload fixed version to ppa I would get that.

**Comment by LDVSOFT** [comment]
Okay, it was Debian package toolchain that reverted the patch, patch does help on Ubuntu 18.10. As I will find out how to publish it I will post here.

**Comment by LDVSOFT** [comment]
Thanks everyone, patched `mutter` for Ubuntu 18.10 is available from [my PPA](https://launchpad.net/~lapshin-dv/+archive/ubuntu/ppa), if one doesn't want to build it manually.

Just do
```
sudo add-apt-repository ppa:lapshin-dv/ppa
sudo apt-get update
sudo apt-get upgrade
```
and you will have a fix. I promise to update the patched version until Ubuntu team releases the official fix.

**Comment by LDVSOFT** [comment]
Ubuntu bug reported, vote to be fixed faster ;) https://bugs.launchpad.net/ubuntu/+source/mutter/+bug/1814169

**Comment by mikehardy** [comment]
@LDVSOFT thanks for the fix and the PPA - I can confirm my workspace transition animations are working correctly with Ubuntu 18.10 + gnome-shell 3.30.1 and your PPA

**Comment by mrmoss** [comment]
I'm 99% sure there's a better way to do this, and it sounds like the fixes to mutter will probably make it unneeded, but I figured I'd share this script. Every time I update my Arch install I have to redo the patch detailed in the guide by @hkdb. This script hasn't failed me yet, and while it's a bit hacky, I want to say it's not the worst I've seen?
[fix_workspace_movement.zip](https://github.com/zakkak/workspace-grid/files/2923562/fix_workspace_movement.zip)

**Comment by LDVSOFT** [comment]
Looks like Ubuntu with 3.30.2 is working without the fix. If anyone will find it different don't hesitate contacting me to rebuild.

**Comment by OlivierBondu** [comment]
@LDVSOFT I find it different :smile: I'm on Ubuntu 18.10 with mutter 3.30.2, but the animations are still not correct ...

however I see in the ubuntu bug tracker this should be fixed with mutter 3.31.90, not 3.30.2 ?

**Comment by LDVSOFT** [comment]
Sorry, it looks like mine version is actually already based on mutter 3.30.2 (my version string is `3.30.2-1~ubuntu18.10.3ldv0`). Hopefully next version will be better, I will rebuild currently published mutter with fix on the weekend.

**Comment by radubuciuceanu** [comment]
Using Ubuntu 18.10 with Gnome 3.30.1. This one helped me: [click](https://github.com/zakkak/workspace-grid/issues/104#issuecomment-457929585). Thx you @LDVSOFT.

**Comment by LDVSOFT** [comment]
Uploaded new version based on `…ubuntu18.10.4`.
